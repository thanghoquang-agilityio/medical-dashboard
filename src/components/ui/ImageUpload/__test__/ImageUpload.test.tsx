import { render, screen } from '@testing-library/react';
import { JSX, ClassAttributes, ImgHTMLAttributes } from 'react';

import { ImageUpload } from '..';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (
    props: JSX.IntrinsicAttributes &
      ClassAttributes<HTMLImageElement> &
      ImgHTMLAttributes<HTMLImageElement>,
  ) => {
    return <img {...props} />;
  },
}));

describe('ImageUpload Component', () => {
  const mockOnRemoveImage = jest.fn();
  const mockOnClick = jest.fn();
  const mockOnUploadImage = jest.fn();

  const renderComponent = (props = {}) =>
    render(
      <ImageUpload
        onRemoveImage={mockOnRemoveImage}
        onUploadImage={mockOnUploadImage}
        onClick={mockOnClick}
        {...props}
      />,
    );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default props', () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('renders an avatar if src is provided', () => {
    renderComponent({ src: '/avatar.png' });
    const avatar = screen.getByAltText('Image for avatar');
    expect(avatar).toHaveAttribute('src', 'http://localhost:1341/avatar.png');
  });

  it('renders an uploaded image if srcUpload is provided', () => {
    renderComponent({ srcUpload: 'uploaded-image.png' });
    const uploadedImage = screen.getByAltText('Image for avatar');
    expect(uploadedImage).toHaveAttribute('src', 'uploaded-image.png');
  });
});
