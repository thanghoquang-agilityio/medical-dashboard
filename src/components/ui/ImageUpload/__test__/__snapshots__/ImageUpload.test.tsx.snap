// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUpload Component renders correctly with default props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col justify-center items-center"
      >
        <div
          class="relative rounded-full"
          style="width: 100px; height: 100px;"
        >
          <label
            class="group cursor-pointer relative"
            for="avatar"
          >
            <span
              class="box-border overflow-hidden outline-none rounded-full ring-offset-background dark:ring-offset-background-dark shrink-0 min-w-8 w-8 h-8 bg-undefined ring-offset-0 ring-0 false"
              tabindex="-1"
            >
              <img
                alt="Image for avatar"
                blurdataurl="/images/avatar-thumbnail.webp"
                class="w-[100px] h-[100px] rounded-full aspect-square"
                height="100"
                src="/images/avatar-thumbnail.webp"
                style="object-fit: cover;"
                width="100"
              />
            </span>
          </label>
        </div>
        <div
          class="group flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] hidden"
          data-filled="true"
          data-filled-within="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            class="flex flex-col h-20"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-transparent border-text-foreground !bg-transparent data-[hover=true]:!bg-transparent group-data-[focus=true]:ring-0 group-data-[focus=true]:ring-offset-0 group-data-[focus=true]:!bg-transparent data-[focus=true]:border-secondary-300 group-data-[invalid=true]:border-danger-100 h-auto py-4 border"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  accept="image/*"
                  aria-label=" "
                  class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-primary-100 placeholder:text-primary-300 bg-transparent text-sm mx-2 group-data-[invalid=true]:!text-danger-100"
                  data-slot="input"
                  data-testid="upload-image"
                  id="avatar"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col justify-center items-center"
    >
      <div
        class="relative rounded-full"
        style="width: 100px; height: 100px;"
      >
        <label
          class="group cursor-pointer relative"
          for="avatar"
        >
          <span
            class="box-border overflow-hidden outline-none rounded-full ring-offset-background dark:ring-offset-background-dark shrink-0 min-w-8 w-8 h-8 bg-undefined ring-offset-0 ring-0 false"
            tabindex="-1"
          >
            <img
              alt="Image for avatar"
              blurdataurl="/images/avatar-thumbnail.webp"
              class="w-[100px] h-[100px] rounded-full aspect-square"
              height="100"
              src="/images/avatar-thumbnail.webp"
              style="object-fit: cover;"
              width="100"
            />
          </span>
        </label>
      </div>
      <div
        class="group flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] hidden"
        data-filled="true"
        data-filled-within="true"
        data-has-value="true"
        data-slot="base"
      >
        <div
          class="flex flex-col h-20"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-transparent border-text-foreground !bg-transparent data-[hover=true]:!bg-transparent group-data-[focus=true]:ring-0 group-data-[focus=true]:ring-offset-0 group-data-[focus=true]:!bg-transparent data-[focus=true]:border-secondary-300 group-data-[invalid=true]:border-danger-100 h-auto py-4 border"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                accept="image/*"
                aria-label=" "
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text group-data-[has-value=true]:text-default-foreground text-primary-100 placeholder:text-primary-300 bg-transparent text-sm mx-2 group-data-[invalid=true]:!text-danger-100"
                data-slot="input"
                data-testid="upload-image"
                id="avatar"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
